e2e/scripts/check-postupgrade-state.sh

```diff
@@ -47,25 +54,29 @@ main() {
         exit 1
     fi
 
-    # ensure that nginx-ingress has been updated
-    # TODO (@salah): fix this
-    # kubectl describe chart -n kube-system k0s-addon-chart-ingress-nginx
-    # # ensure new values are present
-    # if ! kubectl describe chart -n kube-system k0s-addon-chart-ingress-nginx | grep -q "test-upgrade-value"; then
-    #     echo "test-upgrade-value not found in ingress-nginx chart"
-    #     exit 1
-    # fi
-    # # ensure new version is present
-    # if ! kubectl describe chart -n kube-system k0s-addon-chart-ingress-nginx | grep -q "4.12.0-beta.0"; then
-    #     echo "4.12.0-beta.0 not found in ingress-nginx chart"
-    #     exit 1
-    # fi
+    # ensure that the nginx-ingress chart has been updated
+    local latest_nginx_chart_secret=
+    latest_nginx_chart_secret=$(kubectl get secrets -n ingress-nginx | grep helm | awk '{ print $1 }' | sort -r | head -n 1)
+    if ! kubectl get secret -n ingress-nginx "$latest_nginx_chart_secret" -ojsonpath="{.data.release}" | base64 -d | base64 -d | gzip -d | grep -q "test-upgrade-value" ; then
```

Comment:
do we really need this if we're already checking the applied service below?

Reply:
I could go either way honestly